src/errors.rs

```diff
@@ -0,0 +1,123 @@
+use crate::{
+    protos::coresdk::activity_result::ActivityResult,
+    protos::coresdk::workflow_completion::WfActivationCompletion,
+    protos::temporal::api::workflowservice::v1::PollWorkflowTaskQueueResponse,
+    workflow::WorkflowError,
+};
+use tonic::codegen::http::uri::InvalidUri;
+
+pub(crate) struct ShutdownErr;
+pub(crate) struct WorkflowUpdateError {
+    /// Underlying workflow error
+    pub source: WorkflowError,
+    /// The run id of the erring workflow
+    pub run_id: String,
+}
+
+/// Errors thrown during initialization of [crate::Core]
+#[derive(thiserror::Error, Debug, displaydoc::Display)]
+pub enum CoreInitError {
+    /// Invalid URI: {0:?}
+    InvalidUri(#[from] InvalidUri),
+    /// Server connection error: {0:?}
+    TonicTransportError(#[from] tonic::transport::Error),
+}
+
+/// Errors thrown by [crate::Core::poll_workflow_task]
+#[derive(thiserror::Error, Debug, displaydoc::Display)]
+pub enum PollWfError {
+    /// There was an error specific to a workflow instance with id ({run_id}): {source:?}
+    WorkflowUpdateError {
+        /// Underlying workflow error
+        source: WorkflowError,
+        /// The run id of the erring workflow
+        run_id: String,
+    },
+    /// Poll workflow response from server was malformed: {0:?}
+    BadPollResponseFromServer(PollWorkflowTaskQueueResponse),
+    /** [crate::Core::shutdown] was called, and there are no more replay tasks to be handled. You
+     * must call [crate::Core::complete_workflow_task] for any remaining tasks, and then may
+     * exit.*/
+    ShuttingDown,
```

Comment:
Just to clarify, does "no more replay tasks" mean all pending completions have been received from lang and it's safe to stop polling? If this is the case, we should rename to `ShutDown`.

Reply:
Yes, it means that